### PR TITLE
Final `o-table` changes.

### PR DESCRIPTION
**Pull request by notlee**
- Make all mixins except `oTable` private -- accepts a feature flags in an `$opts` argument.
- Removes the default bottom margin, so the margin can be added or not according to the context of the table.
- Update the pa11y demo.
- Update the cell formatter to avoid the "Array.prototype.findIndex" polyfill.

Please review the readme carefully to make sure it all makes sense. :D

### CODE REVIEW

**Comment by notlee** [comment]
Feel unsure about removing the default margin. https://github.com/Financial-Times/o-table/pull/117/commits/2489585fe6f49485d826a24960e1fbddbff9d1dd

If we had some spacing helper classes though this would feel better...

File: README.md [hunk 1]
```diff
@@ -490,11 +479,16 @@ Known issues:
...
+		}
+	```
 - Mixin updates:
-	- `oTableCaptionTop`, `oTableCaptionBottom` have been removed. Use only `oTableCaption` instead.
-	- `oTableCellNumeric`, `oTableCellContentSecondary`, `oTableCellVerticallyCenter`, `oTableCaption` have been removed. Their styles are now included as part of `oTableBase`.
-	- `oTableAll`: does not accept a class name `$classname`. Instead use the default `o-table` class name -- please [contact us](#contact) if this is a problem for your team.
-	- All mixins now output classes directly, so must not be wrapped in an `.o-table` class manually.
+	- All `o-table` mixins have been made private except for a new mixin `@include oTable($opts)`. It accepts an feature list `$opts` to include `o-table` styles granularly. Replace previous mixins with one call to the [`oTable` mixin with an optional `$opts` flag](#silent-mode). Please [contact us](#contact) if this does not suit you product.
+	- `oTableAll` has been replaced with `oTable`, which does not accept a class name `$classname`. Instead use the default `o-table` class name. As the mixin now output classes directly, they must not be wrapped in an `.o-table` class manually.
```
Comment (gvonkoss) [inline README.md]:
```suggestion
	- `oTableAll` has been replaced with `oTable`, which does not accept a class name `$classname`. Instead use the default `o-table` class name in your markup. As the mixin now output classes directly, they must not be wrapped in an `.o-table` class manually.
```

File: README.md [hunk 2]
```diff
@@ -490,11 +479,16 @@ Known issues:
...
+		.o-table {
+			margin-bottom: oTypographySpacingSize($units: 4);
+		}
+	```
 - Mixin updates:
-	- `oTableCaptionTop`, `oTableCaptionBottom` have been removed. Use only `oTableCaption` instead.
-	- `oTableCellNumeric`, `oTableCellContentSecondary`, `oTableCellVerticallyCenter`, `oTableCaption` have been removed. Their styles are now included as part of `oTableBase`.
-	- `oTableAll`: does not accept a class name `$classname`. Instead use the default `o-table` class name -- please [contact us](#contact) if this is a problem for your team.
-	- All mixins now output classes directly, so must not be wrapped in an `.o-table` class manually.
+	- All `o-table` mixins have been made private except for a new mixin `@include oTable($opts)`. It accepts an feature list `$opts` to include `o-table` styles granularly. Replace previous mixins with one call to the [`oTable` mixin with an optional `$opts` flag](#silent-mode). Please [contact us](#contact) if this does not suit you product.
```
Comment (gvonkoss) [inline README.md]:
```suggestion
	- All `o-table` mixins have been made private except for a new mixin `@include oTable($opts)`. It accepts an feature list `$opts` to include `o-table` styles granularly. Replace previous mixins with one call to the [`oTable` mixin with an optional `$opts` flag](#silent-mode). Please [contact us](#contact) if this does not suit your product.
```

File: src/scss/_borders.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /// Horizontal borders
```
Comment (gvonkoss) [inline src/scss/_borders.scss]:
```suggestion
/// Horizontal borders
/// @access private
```

File: src/scss/_borders.scss [hunk 4]
```diff
@@ -11,7 +11,7 @@
 }
 
 /// Vertical borders
```
Comment (gvonkoss) [inline src/scss/_borders.scss]:
```suggestion
/// Vertical borders
/// @access private
```

File: src/scss/_lines.scss [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /// Add horizontal lines to a table
```
Comment (gvonkoss) [inline src/scss/_lines.scss]:
```suggestion
/// Add horizontal lines to a table
/// @access private
```

File: src/scss/_lines.scss [hunk 6]
```diff
@@ -9,7 +9,7 @@
 
 
 /// Add verticle lines to a table
```
Comment (gvonkoss) [inline src/scss/_lines.scss]:
```suggestion
/// Add verticle lines to a table
/// @access private
```

File: src/scss/_mixins.scss [hunk 7]
```diff
@@ -1,36 +1,63 @@
 /// Outputs all features and styles available.
-@mixin oTableAll() {
+/// @param $opts {list} [('responsive-overflow', 'responsive-flat', 'responsive-scroll', 'lines', 'compact', 'stripes', 'row-headings')] - Table features to output. Defaults to all table features.
```
Comment (gvonkoss) [inline src/scss/_mixins.scss]:
```suggestion
/// @param $opts {list} [('responsive-overflow', 'responsive-flat', 'responsive-scroll', 'lines', 'compact', 'stripes', 'row-headings')] - Table features to output. Defaults to all table features.
/// @access public
```

**Pull request merged**